--!strict

export type InteractionConfig = {
	Radius: number,
	TickRateSec: number,
	ServerCooldownSec: number,
	LocalRequestCooldownSec: number,
	UseLineOfSightClient: boolean,
	UseLineOfSightServer: boolean,
	InteractKey: Enum.KeyCode,
	GamepadInteractKey: Enum.KeyCode?,
	PromptText: string,
	ShowInteractId: boolean,
	InteractIdAttribute: string,
	PromptFormatWithId: string,
	PromptSize: UDim2,
	PromptPosition: UDim2,
	PromptPadding: number,
	PromptTextSize: number,
	PromptBillboardSize: UDim2,
	PromptStudsOffset: Vector3,
	PromptTweenTime: number,
	PromptHiddenTransparency: number,
	TargetAcquireTimeSec: number,
	TargetReleaseTimeSec: number,
	DebugEnabled: boolean,
	LogFrequencySec: number,
	SfxSoundId: string,
	SfxVolume: number,
}

local InteractionConfig: InteractionConfig = {
	Radius = 12,
	TickRateSec = 0.1,
	ServerCooldownSec = 0.6,
	LocalRequestCooldownSec = 0.2,
	UseLineOfSightClient = true,
	UseLineOfSightServer = true,
	InteractKey = Enum.KeyCode.E,
	GamepadInteractKey = Enum.KeyCode.ButtonX,
	PromptText = "Press E",
	ShowInteractId = true,
	InteractIdAttribute = "InteractId",
	PromptFormatWithId = "%s  •  %s",
	PromptSize = UDim2.fromOffset(220, 48),
	PromptPosition = UDim2.fromScale(0.5, 0.75),
	PromptPadding = 10,
	PromptTextSize = 14,
	PromptBillboardSize = UDim2.fromOffset(220, 52),
	PromptStudsOffset = Vector3.new(0, 2.5, 0),
	PromptTweenTime = 0.18,
	PromptHiddenTransparency = 1,
	TargetAcquireTimeSec = 0.12,
	TargetReleaseTimeSec = 0.2,
	DebugEnabled = false,
	LogFrequencySec = 1.2,
	SfxSoundId = "rbxassetid://6895079853",
	SfxVolume = 0.7,
}

return InteractionConfig
